src/mode/modeVisual.ts

```diff
@@ -0,0 +1,140 @@
+"use strict";
+
+import * as _      from 'lodash';
+
+import { ModeName, Mode } from './mode';
+import { Motion} from './../motion/motion';
+import { Position } from './../motion/position';
+import { Operator } from './../operator/operator';
+import { DeleteOperator } from './../operator/delete';
+import { ModeHandler } from './modeHandler.ts';
+import { ChangeOperator } from './../operator/change';
+
+export class VisualMode extends Mode {
+    /**
+     * The part of the selection that stays in the same place when motions are applied.
+     */
+    private _selectionStart: Position;
+
+    /**
+     * The part of the selection that moves.
+     */
+    private _selectionStop : Position;
+    private _modeHandler   : ModeHandler;
+
+    private _keysToOperators: { [key: string]: Operator };
+
+    constructor(motion: Motion, modeHandler: ModeHandler) {
+        super(ModeName.Visual, motion);
+
+        this._modeHandler = modeHandler;
+        this._keysToOperators = {
+            // TODO: use DeleteOperator.key()
+
+            // TODO: Don't pass in mode handler to DeleteOperators,
+            // simply allow the operators to say what mode they transition into.
+            'd': new DeleteOperator(modeHandler),
```

Comment:
And then later we turn on `--experimentalDecorators` and this list autopopulates and its magic and unicorns everywhere! :star: :fireworks: :dancers: